### PR TITLE
feat(TCOMP-2084): add i18n-ed metadata

### PR DESCRIPTION
**Pull request by undx**
https://jira.talendforge.org/browse/TCOMP-2084

### CODE REVIEW

File: component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java [hunk 1]
```diff
@@ -570,36 +574,36 @@ private ComponentIndex toComponentIndex(final Container container, final Locale
...
-                        // yet
-                        .map(category -> meta
-                                .getParent()
-                                .findBundle(loader, locale)
-                                .category(category)
-                                .orElseGet(() -> category
-                                        .replace("/" + meta.getParent().getName() + "/",
-                                                "/" + familyDisplayName + "/")))
+                        .map(category -> category.replace("${family}", meta.getParent().getName()))
+                        // not i18n-ed yet
```
Comment (ozhelezniak-talend) [inline component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:585]: I think this comment should be above the line 584
.map(category -> category.replace("${family}", meta.getParent().getName()))
it was related to it.
Now it looks like it comments the 586

File: component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java [hunk 2]
```diff
@@ -610,6 +614,18 @@ private String normalizeCategory(final String category) {
         return category;
     }
 
+    private Map<String, String> translateMetadata(final Map<String, String> source, final ComponentBundle bundle) {
+        return source
+                .entrySet()
+                .stream()
+                .map(e -> {
+                    final String bundleKey = e.getKey().replaceAll("::", ".");
+                    e.setValue(bundle.displayName(bundleKey).orElse(e.getValue()));
```
Comment (ozhelezniak-talend) [inline component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:623]: Is it okay to change the internal state of the in-parameter source?
I see ".orElse()" part will back us up, but still. 
From the javadoc of setValue:
> Replaces the value corresponding to this entry with the specified value (optional operation). (Writes through to the map.)

**Review by ozhelezniak-talend — COMMENTED** [review]
left a few comments

File: component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java (hunk 1 above)
Comment (undx) [reply component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:585]: right! will simply remove comment, it's useless.

File: component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java [hunk 3]
```diff
@@ -610,6 +614,21 @@ private String normalizeCategory(final String category) {
         return category;
     }
 
+    private Map<String, String> translateMetadata(final Map<String, String> source, final ComponentBundle bundle) {
+        return source
+                .entrySet()
+                .stream()
+                .map(e -> {
+                    final String bundleKey = e.getKey().replaceAll("::", ".");
```
Comment (ozhelezniak-talend) [inline component-server-parent/component-server/src/main/java/org/talend/sdk/component/server/front/ComponentResourceImpl.java:622]: Not 100% sure that it's better but anyway:

```java
Optional.of(e.getKey().replaceAll("::", "."))
.flatMap(it->bundle.displayName(it))
.map(it->new SimpleEntry(e.getKey, it))
.orElse(e)
```

**Review by ozhelezniak-talend — APPROVED** [review]
LGTM 
(minor comment: just sugggestion)

**Review by ozhelezniak-talend — APPROVED** [review]
LGTM

**Pull request merged**